### PR TITLE
Reuse ProcessMessageSubscriptionRecord in ProcessMessageSubscription state

### LINKED ISSUE #6533: Reuse the process instance subscription record in the state
**Description**

Currently, the `ProcessInstanceSubscriptionState` persists the subscriptions as `ProcessInstanceSubscription` objects. These objects are almost equal to the `ProcessInstanceSubscription` record. We should reuse the record in the state to avoid duplication and reloading the subscription in the event applier.   

Follow-up of #6182.

### PR DESCRIPTION
**Pull request by npepinpe**
## Description

This PR reduces duplication between `ProcessMessageSubscription` and `ProcessMessageSubscriptionRecord`. It does so by embedding the record into the subscription object.

Additionally, it reduces the API surface of the `MutableProcessMessageSubscriptionState`, which is possible due to the way we now embed the record into the state object.

NOTE: there are two formatting commits which are plain automated formatting according to the ZeebeStyle, you can skip those. As well there is one commit which is just updating the test to the new object format, but the tests themselves weren't changed. The interesting commits are then, in order: 
- https://github.com/camunda-cloud/zeebe/commit/a249792ea57202227cef4f1abbdffd14b2e1cf53 (new subscription object)
- https://github.com/camunda-cloud/zeebe/commit/5d0060a8e3f7437a5bf59bfcaf146877d873b737 (refactor state to use new object)
- https://github.com/camunda-cloud/zeebe/commit/a4ec51617c4913d7d1ea54a15e1a0c9e42bec5b7 (update usages of the state and the object)

## Related issues

closes #6533

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/0.25`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

**Comment by npepinpe** [comment]
Seems some things are broken (though I could've sworn I had fixed everything :thinking:) - that'll teach me to request a review too early. I'll ping you again when it's ready.

File: engine/src/main/java/io/zeebe/engine/processing/message/ProcessMessageSubscriptionCorrelateProcessor.java [hunk 1]
```diff
@@ -104,23 +103,25 @@ public void processRecord(
...
-        // TODO (saig0): reuse the subscription record in the state (#6533)
-        subscriptionRecord
-            .setElementId(subscription.getTargetElementId())
-            .setInterrupting(subscription.shouldCloseOnCorrelate());
+        // avoid reusing the subscription record directly as any access to the state (e.g. as #get)
+        // will overwrite it - safer to just copy its values into an one-time-use record
+        final ProcessMessageSubscriptionRecord subscriptionRecord = subscription.getRecord();
+        record
+            .setElementId(subscriptionRecord.getElementIdBuffer())
+            .setInterrupting(subscriptionRecord.isInterrupting());
```
Comment (npepinpe) [inline engine/src/main/java/io/zeebe/engine/processing/message/ProcessMessageSubscriptionCorrelateProcessor.java:112]: I imagine from the TODO this was not the original plan - however, I think it's safer to do it like this. It's tricky that accessing the state changes the record (for example, in the appliers), very easy to forget. I think in the long term it makes sense to use the record stored in the state as a template in the subscription message handler, but not here, such that the record you get here would already contain all the info you need, and you wouldn't need to fetch the subscription here.

We can do that once subscriptions have keys I guess and we refactor the message handler/API.

**Review by saig0 — APPROVED** [review]
Nice work :cake:

**Comment by npepinpe** [comment]
bors retry

**Pull request merged**